tests/functional/test_snap_dcgm.py

```diff
@@ -0,0 +1,76 @@
+import json
+import subprocess
+import urllib.request
+from time import sleep
+
+from tenacity import retry, stop_after_delay, wait_fixed
+
+
+@retry(wait=wait_fixed(5), stop=stop_after_delay(15))
+def test_dcgm_exporter():
+    """Test of the dcgm-exporter service and its endpoint."""
+    endpoint = "http://localhost:9400/metrics"
+    dcgm_exporter_service = "dcgm.dcgm-exporter"
+
+    subprocess.run(f"sudo snap start {dcgm_exporter_service}".split(), check=True)
+
+    result = subprocess.run(
+        f"sudo snap services {dcgm_exporter_service}".split(),
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+    assert " active" in result.stdout.strip(), "dcgm-exporter service is not active"
+
+    urllib.request.urlopen(endpoint)
```

Comment:
please add a comment here to explain why this is here - I'm guessing it will raise an exception if it's not listening?  Can we make any assertions about the contents of the response?

Reply:
We could, but it will only have non-empty content when running on the hardware. If it's a VM environment, there will be no content as there are no Nvidia GPUs.

Reply:
right, but I think being more explicit would be beneficial. You could:
* add a comment explaining that we don't check the output because we're testing in a virtualized environment and the exporter does not generate any metric in that environment
* use an assertion to clarify what exactly we're looking for (a 200)

Reply:
Thanks @Deezzir :)